### ISSUE TITLE
function definitions in K&R style of C are no longer recognized

### ISSUE THREAD

**Issue by vladak** [universal-ctags/ctags#1306]
**Test file:**

named `foo.c`:

```
size_t
foo(as)
        register struct as *as;
{
        if (foo)
                return (0);
}

ushort_t
bar(struct as *as)
{
        return (bar);
}
```

**ctags version**:

```
Universal Ctags 0.0.0(d2efd07), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Feb 17 2017, 09:20:27
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +option-directory, +coproc, +xpath, +json, +interactive
```

**actual result from universal ctags**

```
!_TAG_FILE_FORMAT       2       /extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED       1       /0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_MODE       u-ctags /u-ctags or e-ctags/
!_TAG_PROGRAM_AUTHOR    Universal Ctags Team    //
!_TAG_PROGRAM_NAME      Universal Ctags /Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL       https://ctags.io/       /official site/
!_TAG_PROGRAM_VERSION   0.0.0   /d2efd07/
as      /tmp/foo.c      /^      register struct as *as;$/;"     v       typeref:struct:as *
bar     /tmp/foo.c      /^bar(struct as *as)$/;"        f       typeref:typename:ushort_t
```

**exuberant ctags**

```
ctags: Warning: Ignoring non-option in ./.ctags

Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Feb 21 2012, 13:56:11
  Addresses: <dhiebert@users.sourceforge.net>, http://ctags.sourceforge.net
  Optional compiled features: +wildcards, +regex
```

**actual result from exuberant ctags**

```
!_TAG_FILE_FORMAT       2       /extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED       1       /0=unsorted, 1=sorted, 2=foldcase/
!_TAG_PROGRAM_AUTHOR    Darren Hiebert  /dhiebert@users.sourceforge.net/
!_TAG_PROGRAM_NAME      Exuberant Ctags //
!_TAG_PROGRAM_URL       http://ctags.sourceforge.net    /official site/
!_TAG_PROGRAM_VERSION   5.8     //
bar     /tmp/foo.c      /^bar(struct as *as)$/;"        f
foo     /tmp/foo.c      /^foo(as)$/;"   f
```

**Comment by masatake** [comment]
Thank you for reporing.

```
[jet@localhost ctags]$ cat b.c
void foo(x) struct as *x; {}
[jet@localhost ctags]$ cat g.c
void foo(x) int *x; {}
[jet@localhost ctags]$ ./ctags -o - b.c
x	b.c	/^void foo(x) struct as *x; {}$/;"	v	typeref:struct:as *
[jet@localhost ctags]$ ./ctags -o - g.c
foo	g.c	/^void foo(x) int *x; {}$/;"	f
```

It seems that `struct` in parameter declaration block handled wrongly.

**Comment by pragmaware** [comment]
Fixed via #1314
